app/src/main/java/com/nextcloud/talk/adapters/messages/IncomingLinkPreviewMessageViewHolder.kt

```diff
@@ -92,6 +97,16 @@ class IncomingLinkPreviewMessageViewHolder(incomingView: View, payload: Any) :
 
         binding.messageText.text = processedMessageText
 
+        val link = SpannableString(processedMessageText)
```

Comment:
`processedMessageText` can be mixed text with hyperlinks. So "test http://anysite.com" can't be directly converted to a link but the link has to be extracted and only this should be underlined and clickable.
Otherwise there could be a ActivityNotFoundException if the link is not valid.

Reply:
to extract links, code from https://github.com/nextcloud/talk-android/blob/7ca68b334a625e82a3865b5d1a8d7d50343a5cfe/app/src/main/java/com/nextcloud/talk/chat/data/model/ChatMessage.kt#L167
will be helpful. This is the code which decides if the *LinkPreviewMessageViewHolder is applied.

Reply:
However thinking about that there can be multiple links inside a message this gets more complicated. Then there might have to be more complex logic similar like it's used for the chip mentions: https://github.com/nextcloud/talk-android/blob/9d62a6f7456d5980433a1e4d58a35e2f4dbbccdb/app/src/main/java/com/nextcloud/talk/utils/DisplayUtils.kt#L220